dotnet/Vaas/src/Vaas/Vaas.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.WebSockets;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Vaas.Messages;
+using Websocket.Client;
+using Websocket.Client.Exceptions;
+
+namespace Vaas
+{
+    public class Vaas
+    {
+        private string Token { get; }
+
+        private WebsocketClient? Client { get; set; }
+
+        private string? SessionId { get; set; }
+
+        private bool Authenticated { get; set; }
+        
+        public Uri Url { get; set; } = new("wss://gateway-vaas.gdatasecurity.de");
+
+        private Dictionary<string, VerdictResponse> VerdictResponsesDict { get; } = new();
+
+        public Vaas(string token)
+        {
+            Token = token;
+        }
+
+        public void Connect()
+        {
+            Client = new WebsocketClient(Url, CreateWebsocketClient());
+            Client.ReconnectTimeout = null;
+            Client.MessageReceived.Subscribe(msg =>
+            {
+                if (msg.MessageType != WebSocketMessageType.Text) return;
+                var message = JsonSerializer.Deserialize<Message>(msg.Text);
+                switch (message?.Kind)
+                {
+                    case "AuthResponse":
+                        var authenticationResponse = JsonSerializer.Deserialize<AuthenticationResponse>(msg.Text);
+                        if (authenticationResponse is {Success: true})
+                        {
+                            Authenticated = true;
+                            SessionId = authenticationResponse.SessionId;
+                        }
+                        break;
+                        
+                    case "VerdictResponse":
+                        var options = new JsonSerializerOptions {Converters = {new JsonStringEnumConverter()}};
+                        var verdictResponse = JsonSerializer.Deserialize<VerdictResponse>(msg.Text,options);
+                        VerdictResponsesDict.Add(verdictResponse?.Guid ?? throw new InvalidOperationException(), verdictResponse);
+                        break;
+                }
+            });
+            Client.Start().GetAwaiter().GetResult();
+            if (!Client.IsStarted)
+            {
+                throw new WebsocketException("Could not start client");
+            }
+            
+            Authenticate();
+            Task.Run(SendPing);
+        }
+
+        private void Authenticate()
+        {
+            var authenticationRequest = new AuthenticationRequest(Token);
+            var jsonString = JsonSerializer.Serialize(authenticationRequest);
+            Client?.Send(jsonString);
+            for (var i = 0; i < 10; i++)
+            {
+                if (Authenticated)

```

Comment:
Updating a bool from another thread may not become visible (if the variable gets optimized into a register). This was the case in the past with x86 release builds. This works for x64 builds, but is not guaranteed by the memory model.

tldr: This might break in the future.

Reply:
Switched to Task.